tutorials/sphinx_tuto/tensorclass_imagenet.py

```diff
@@ -153,59 +250,132 @@ def __len__(self):
 # ``__getitems__``) we can use them like a map-style Dataset and create a
 # ``DataLoader`` directly from them.
 #
-# Note that because the tensorclass can handle batched indices, there is
-# no need for additional collation, so we pass the identity function as
-# the ``collate_fn``.
+# Since the TensorClass data will be loaded in batches, we need to specify how these
+# batches should be collated. For this we write the following helper class
+
+
+class Collate(nn.Module):
+    def __init__(self, transform=None, device=None):
+        super().__init__()
+        self.transform = transform
+        self.device = device
+
+    def __call__(self, x: ImageNetData):
+        # move data to RAM
+        out = x.apply(lambda x: x.contiguous())
+        if self.device:
+            # move data to gpu
+            out = out.to(self.device)
+        if self.transform:
+            # apply transforms on gpu
+            out.images = self.transform(out.images)
+        return out
 
-batch_size = 64
-train_dataloader = DataLoader(train_data, batch_size=batch_size)
-val_dataloader = DataLoader(val_data, batch_size=batch_size)
+
+##############################################################################
+# ``DataLoader`` has support for multiple workers loading data in parallel. The
+# tensorclass dataloader will use just one worker, but load data in batches.
+
+batch_size = 8
```

Comment:
On the cluster I used 128 and it worked fine

Reply:
Just saw that you corrected that below :)

Reply:
Yeah, 128 works great for the full imagenet, but caused some problems in the benchmarking code on the hymenoptera data (`t0` is only defined after certain number of batches which was not reached).